### ISSUE TITLE
FULL names not staying attached to their combinations

### ISSUE THREAD

**Issue by Landru101** [TES5Edit/TES5Edit#1401]
**What is the version of xEdit you are using?**
FO4Edit 4.1.5f (AC5635A7) starting session 2024-12-20 18:01:26

**Describe the bug**
When I add a FULL name to a combination in an object template (ARMO record), it shows up on a different combination after I save & restart FO4Edit.

**To reproduce**
Two examples:
First: Move the FULL name to the correct combination, and xEdit says the file hasn't changed.
This is just an example to show some unexpected behavior. Normallly I wouldn't edit this file AT ALL, EVER.
I'm working on compatability patches for Horizon 1.9.4d
https://www.nexusmods.com/fallout4/mods/17374
Z_Horizon.esp: 00140C3D ARMO Armor_Power_T60_ArmLeft
Notice that the "Ruined" FULL name is on combination 14 (BOSSentinel), but -- IMHO -- it should be on 15 (if_tmp_armor_RuinedCondition).
1. Remove the FULL name from Combination 14.
2. Add the FULL name property to Combination 15.
3. Edit it to say "Ruined"
4. Save.
5. Observe the message "File has not changed, removing: Z_Horizon.esp.save.2024_12_20_17_56_20"

Second: Do the same thing in a new ESP and watch it float up one combination after you save & restart xEdit.
1. Copy that record 00140C3D from Z_Horizon.esp to a new ESL-flagged ESP.
2. Perform the same operations above in the new plugin.
3. Save.
4. Exit xEdit and restart.
5. Observe the "Ruined" name on Combination 14 and not 15 where you just added it.

**Expected behavior**
I'm new to object templates and combinations, so maybe this is not a bug, but, example 1: if you change the file, xEdit should recognize that the file has changed; example 2: if you add a FULL name to a combination, that name should come up on the same combination after you save your work and restart xEdit. I mean, I don't know, maybe every single combination before the one you want to edit has to have a FULL name property, and it has to be non-null, for it to work. It may just be how the record format / file format works.

**Screenshots**
The attached screenshot shows the Ruined name on combination 14 after I just added it to combination 15 and restarted xEdit.
![FULLNameTest](https://github.com/user-attachments/assets/9f43cb84-3186-4c82-8e8a-ebfefc725500)

**Additional context**
(What I was ultimately trying to do was make a Horizon compatability patch for another mod (BPAO) that adds more BoS armor combinations. Got sidetracked by this behavior.)
I don't know how I can get that name onto the right keyword and have it stay there. The answer may be the typical Internet thing where someone says I want to do X, and everyone says, WELL ACKSHUALLY no, you don't want to do X. Maybe you cannot add or change combinations without breaking everything.
I've searched and haven't been able to find anything on topic. See generally https://forums.nexusmods.com/topic/6758452-what-is-best-for-item-object-templates-copying-an-item-and-adding-a-new-ot-or-adding-an-ot-to-a-vanilla-item/
(edited to remove number signs)

**Comment by Kuroitsune** [comment]
I think that's because Combinations are aligned. If 14 and 15 are exactly the same aside from FULL being different, then when you swap the names, xEdit isn't going to save that change, because you didn't really change anything, simply moved data in an array that is aligned. Same for saving into a new file and doing the same swap.

That's if I'm right, I could be wrong. Combinations are done through wbRArray and wbRStruct which I'm sure are aligned.

**Comment by robertgk2017** [comment]
Yes that's dueto Array alignment which is the default behavior of all Arrays.  the order will still match what's written in the file, but if the order is different across overrides it'll gap/spread out the array vertically to make things line up across overrides.

**Comment by Landru101** [comment]
Where can I read more about this? Sorry for asking, it's just not clicking for me. It seems like this alignment is different from C struct alignment & padding. If I should just use the source, that's fine, I can do that.

(The example in 18.8.51 - Optional "Align alignable array elements" could use a legend or labels on the axes. It wasn't helpful for me.)

**Comment by robertgk2017** [comment]
Is this still an issue in current 4.1.5o?

Edit: Just confirmed that its not.  Closing as Completed.
